pkg/mapi/gcp.go

```diff
@@ -0,0 +1,111 @@
+		if CurrentSpecReport().State == gotypes.SpecStateSkipped {
+			return
+		}
+		// Clean up MAPI MachineSets
+		if mapiMachineSet != nil {
+			err := framework.DeleteMachineSets(cl, mapiMachineSet)
+			Expect(err).ToNot(HaveOccurred(), "Failed to delete MAPI MachineSet")
+			framework.WaitForMachineSetsDeleted(ctx, cl, mapiMachineSet)
+		}
+	})
+
+	It("should have all Shielded VM options disabled when using nonUefi image", func() {
+		// Get MAPI machineset parameters
+		machineSetParams := framework.BuildMachineSetParams(ctx, cl, 1)
+
+		// Override the name to include testcaseid 83064
+		infra, err := framework.GetInfrastructure(ctx, cl)
+		Expect(err).NotTo(HaveOccurred(), "Failed to get cluster infrastructure object")
+		Expect(infra.Status.InfrastructureName).ShouldNot(BeEmpty(), "infrastructure name was empty on Infrastructure.Status.")
+		machineSetParams.Name = infra.Status.InfrastructureName + "-83064-" + uuid.New().String()[0:5]
+
+		// Modify the provider spec to use the nonUefi image
+		nonUefiImage := "projects/redhat-marketplace-public/global/images/redhat-coreos-ocp-48-x86-64-202210040145"
+
+		// Unmarshal the provider spec to modify it
+		providerSpec := &mapiv1.GCPMachineProviderSpec{}
+		Expect(json.Unmarshal(machineSetParams.ProviderSpec.Value.Raw, providerSpec)).To(Succeed(), "Should be able to unmarshal provider spec")
+
+		// Set the specific image
+		providerSpec.Disks[0].Image = nonUefiImage
+
+		// Marshal back to raw bytes using JSON
+		rawProviderSpec, err := json.Marshal(providerSpec)
+		Expect(err).ToNot(HaveOccurred(), "Should be able to marshal provider spec")
+
+		machineSetParams.ProviderSpec.Value = &runtime.RawExtension{
+			Raw: rawProviderSpec,
+		}
+
+		By("Creating a new MachineSet with Red Hat CoreOS(nonUefi) image")
+		mapiMachineSet, err = framework.CreateMachineSet(cl, machineSetParams)
+		Expect(err).ToNot(HaveOccurred(), "MachineSet should be able to be created")
+
+		framework.WaitForMachineSet(ctx, cl, mapiMachineSet.GetName())
+
+		By("Verifying that all Shielded VM options are disabled")
+		// Get the machines created by this MachineSet
+		machines, err := framework.GetMachinesFromMachineSet(ctx, cl, mapiMachineSet)
+		Expect(err).ToNot(HaveOccurred(), "Getting machine from MachineSet should succeed")
+
+		// Get the first machine created by this MachineSet and verify its provider spec
+		machine := machines[0]
+		By(fmt.Sprintf("Getting machine %q created by MachineSet %q", machine.Name, mapiMachineSet.Name))
+
+		// Unmarshal the machine's provider spec to check Shielded VM configuration
+		machineProviderSpec := &mapiv1.GCPMachineProviderSpec{}
+		Expect(json.Unmarshal(machine.Spec.ProviderSpec.Value.Raw, machineProviderSpec)).To(Succeed(), "Should be able to unmarshal machine provider spec")
+
+		// Verify that ShieldedInstanceConfig exists and all options are disabled
+		Expect(machineProviderSpec.ShieldedInstanceConfig).ToNot(BeNil(), "ShieldedInstanceConfig should exist")
```

Comment:
Nit; we might be able to do a `HaveField` and `SatisfyAll` for these, maybe easier to read?